meterfs/platform/test_file_phx.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Meterfs test file abstraction
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Aleksander Kaminski, Andrzej Glowinski, Tomasz Korniluk
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <sys/msg.h>
+#include <string.h>
+#include <unistd.h>
+#include <meterfs.h>
+
+#include "test_file.h"
+
+static oid_t meterfs;
+
+static inline void test_filePrepareDevCtl(msg_t* msg)
+{
+	msg->type = mtDevCtl;
+	msg->i.data = NULL;
+	msg->i.size = 0;
+	msg->o.data = NULL;
+	msg->o.size = 0;
+}
+
+
+int test_fileLookup(const char *name)
+{
+	msg_t msg;
+
+	return lookup(name, &msg.i.openclose.oid, NULL);
+}
+
+
+int test_fileOpen(const char *name)
+{
+	msg_t msg;
+	int err;
+	int id;
+
+	if ((err = lookup(name, &msg.i.openclose.oid, NULL)) < 0)
+		return err;
+
+	id = msg.i.openclose.oid.id;
+
+	msg.type = mtOpen;
+	msg.i.data = NULL;
+	msg.i.size = 0;
+	msg.o.data = NULL;
+	msg.o.size = 0;
+	msg.i.openclose.flags = 0;
+
+	if ((err = msgSend(meterfs.port, &msg)) < 0)
```

Comment:
I think that error from msgSend should be interpreted as a system error and returns with fail immediately. Also in other functions.

Reply:
Done.